code/modules/reagents/machinery/chem_master.dm

```diff
@@ -162,6 +162,12 @@ var/global/list/pillIcon2Name = list("oblong purple-pink", "oblong green-white",
 		src.updateUsrDialog()
 		return 1
 
+	else if(istype(B, /obj/item/weapon/reagent_containers/pill))
+		var/name = reject_bad_text(input(user,"Name:","Name your pill!","[B.reagents.get_master_reagent_name()] ([B.reagents.total_volume] units)") as null|text)
+		if(name)
+			B.name = "[name] pill"
+		B.icon_state = "pill"+pillsprite
```

Comment:
Probably need to return 1 here to prevent the "You attack the chemmaster with the pill" message

Reply:
Tested, causes no message

Reply:
Should still return `TRUE` though. It's not only about the message.